src/Magnet.cpp

```diff
@@ -13,14 +11,29 @@ Magnet::Magnet(unsigned height, unsigned width)
         spins.push_back(std::vector<int>());    //Puts an empty vector on it
         spins[i].reserve(width);
         for(unsigned j = 0; j < width; j++)
-            spins[i].push_back(distribution(gen) ? 1 : -1);
+            spins[i].push_back(distribution(rng) ? 1 : -1);
     }
 }
 
+Magnet::Magnet(std::vector<std::vector<int>> spins):
+    rng(std::random_device{}())
+{
+    this->spins = spins;
```

Comment:
Note that this is not a deep copy/move - is this purposeful? Such behaviour could be very useful for making tests, but otherwise I have no idea whether it's good or bad. What is your opinion?

Reply:
I disagree, I believe that operator= for std::vector does in fact result in a deep copy.  I have tested this with a minimal program which I can provide to you for convincing if you'd like.  If we were dealing with C-style arrays (effectively just a pointer), I would agree with you.  The initialisation of a magnet should not be something that is done a huge number of times, so I thought that a deep copy (slow but not confusing) would be appropriate here.

Reply:
Oh, interesting, okay, then that's all fine.